pkg/services/librarypanels/database.go

```diff
@@ -229,3 +229,30 @@ func (lps *LibraryPanelService) patchLibraryPanel(c *models.ReqContext, cmd patc
 
 	return libraryPanel, err
 }
+
+func (lps *LibraryPanelService) GetLibraryPanelsForDashboard(query GetLibraryPanelsQuery) (map[string]LibraryPanel, error) {
+	libraryPanelMap := make(map[string]LibraryPanel)
+	err := lps.SQLStore.WithDbSession(context.Background(), func(session *sqlstore.DBSession) error {
+		sql := `SELECT
+				lp.id, lp.org_id, lp.folder_id, lp.uid, lp.name, lp.model, lp.created, lp.created_by, lp.updated, updated_by
+			FROM
+				library_panel_dashboard lpd
+			INNER JOIN
+				library_panel lp ON lpd.librarypanel_id = lp.id AND lpd.dashboard_id=?`
+
+		libraryPanels := make([]LibraryPanel, 0)
```

Comment:
```suggestion
		var libraryPanels []LibraryPanel
```

Reply:
Damn, I never learn when to use which. Thank you for suggesting the change.

Reply:
@hugohaggmark It's just a nuance, but I figure there's no point in allocating an array if it's just going to be replaced.

Reply:
Fixed!